src/ovirtapi/index.js

```diff
@@ -258,7 +260,7 @@ const OvirtApi = {
     assertLogin({ methodName: 'snapshotNics' })
     return httpGet({ url: `${AppConfiguration.applicationContext}/api/vms/${vmId}/snapshots/${snapshotId}/nics` })
   },
-  snapshot ({ vmId, snapshotId }: { vmId: string, snapshotId: string }): Promise<ApiSnapshotType> {
+  snapshot ({ vmId, snapshotId }: { vmId: string, snapshotId: string }): Promise<Object> {
```

Comment:
Because the function could return an error and that isn't covered by `ApiSnapshotType`.  All of the API functions need a better generic/parameterized type that can be either the error response or the actual API object.